server/issue.go

```diff
@@ -20,6 +20,77 @@ import (
 	"github.com/mattermost/mattermost-plugin-jira/server/utils"
 )
 
+func makePost(userId, channelId, message string) *model.Post {
+	return &model.Post{
+		UserId:    userId,
+		ChannelId: channelId,
+		Message:   message,
+	}
+}
+
+func respondf(plugin *Plugin, status int, e error, mmuserID, jiraBotID, channelID, message string) (int, error) {
+	_ = plugin.API.SendEphemeralPost(mmuserID, makePost(jiraBotID, channelID, message))
+	if e != nil {
+		return status, e
+	}
+	return status, errors.New(message)
+}
+
+func httpAPITransitionIssue(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	requestData := model.PostActionIntegrationRequestFromJson(r.Body)
+	plugin := ji.GetPlugin()
+	jiraBotID := plugin.getUserID()
+	channelID := requestData.ChannelId
+
+	mattermostUserId := requestData.UserId
+	if mattermostUserId == "" {
+		return respondf(plugin, http.StatusUnauthorized, nil, mattermostUserId, jiraBotID, channelID, "user not authorized")
+	}
+
+	var issueKey string
+	var toState string
+	var found bool
+	var ok bool
+	var x interface{}
+
+	if x, found = requestData.Context["issueKey"]; found {
+		if issueKey, ok = x.(string); !ok {
+			return respondf(plugin, http.StatusInternalServerError, nil, mattermostUserId, jiraBotID, channelID, "Issue key type is wrong")
+		}
+	} else {
+		return respondf(plugin, http.StatusInternalServerError, nil, mattermostUserId, jiraBotID, channelID, "No issue key was found in context data")
+	}
+
+	if x, found = requestData.Context["selected_option"]; found {
+		if toState, ok = x.(string); !ok {
+			return respondf(plugin, http.StatusInternalServerError, nil, mattermostUserId, jiraBotID, channelID, "Transition option type is wrong")
+		}
+	} else {
+		return respondf(plugin, http.StatusInternalServerError, nil, mattermostUserId, jiraBotID, channelID, "No transition option was found in context data")
+	}
+
+	jiraUser, err := plugin.userStore.LoadJIRAUser(ji, mattermostUserId)
+	if err != nil {
+		return respondf(plugin, http.StatusUnauthorized, err, mattermostUserId, jiraBotID, channelID, "Your username is not connected to Jira. Please type `jira connect`.")
```

Comment:
Please add the missing slash for the slash command

```suggestion
		return respondf(plugin, http.StatusUnauthorized, err, mattermostUserId, jiraBotID, channelID, "Your username is not connected to Jira. Please type `/jira connect`.")
```

Reply:
Worked on this.